From b8984be0c081713b892726359c1be319b2c8ccab Mon Sep 17 00:00:00 2001
From: Umesh Agarwal <umeshx.agarwal@intel.com>
Date: Thu, 23 Aug 2018 03:20:38 +0530
Subject: [PATCH 2/2] Fix for Headset icon shown for reference phone after
 reset

Reason:
- Once we pair & connect reference device to IVI, and
  perform BT Factory reset which disables bluetooth
  service and clear configurations.After completion of
  reset,Bluetooth Service comes up and sends connect
  request to reference phone,but ideally it should not
  have any info of previously paired device.
- In Application,at the time of reset it only disables
  BT Service and clear bt_config file,but nowhere it
  removes bond from paired devices,due to which IVI has
  sent reconnect request to previously paired device.

Fix:
- In Application,at the time of BT Factory reset,called
  removeBonded() for all paired devices. So that it will
  remove bond and then disable the Bluetooth Service.

Testing:
- Connected Pixel2 & Pixel with IVI,triggered reset from
  setting app.As expected, after reset completion there
  was no reconnect request sent for previously paired
  devices.

Tracked-On: OAM-65743

Signed-off-by: Umesh Agarwal <umeshx.agarwal@intel.com>
---
 src/com/android/bluetooth/btservice/AdapterService.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/com/android/bluetooth/btservice/AdapterService.java b/src/com/android/bluetooth/btservice/AdapterService.java
index ad789551..1c31cfa4 100644
--- a/src/com/android/bluetooth/btservice/AdapterService.java
+++ b/src/com/android/bluetooth/btservice/AdapterService.java
@@ -1435,6 +1435,12 @@ public class AdapterService extends Service {
             if (service == null) {
                 return false;
             }
+            BluetoothDevice[] bondedDeviceList = getBondedDevices();
+            if (bondedDeviceList != null) {
+                for (BluetoothDevice device : bondedDeviceList) {
+                    service.removeBond(device);
+                }
+            }
             service.disable();
             return service.factoryReset();
 
-- 
2.17.1

